core/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/DateHistogramAggregationBuilder.java

```diff
@@ -137,9 +139,7 @@ public DateHistogramAggregationBuilder(String name) {
     /** Read from a stream, for internal use only. */
     public DateHistogramAggregationBuilder(StreamInput in) throws IOException {
         super(in, ValuesSourceType.NUMERIC, ValueType.DATE);
-        if (in.readBoolean()) {
```

Comment:
I decided to remove this boolean flag in the (date) histogram agg builders since it is always true (we always have an order). This made the BWC code a bit more complicated (for some reason `InternalHistogram` did not use this flag), but simplified the new read/write logic.